src/AST/SymbolContext.cs

```diff
@@ -97,37 +97,21 @@ public bool FindSymbol(ref string symbol)
             if (FindLibraryBySymbol(symbol, out lib))
                 return true;
 
-            string alternativeSymbol;
-
-            // Check for C symbols with a leading underscore.
-            alternativeSymbol = "_" + symbol;
-            if (FindLibraryBySymbol(alternativeSymbol, out lib))
-            {
-                symbol = alternativeSymbol;
-                return true;
-            }
-
-            alternativeSymbol = symbol.TrimStart('_');
-            if (FindLibraryBySymbol(alternativeSymbol, out lib))
-            {
-                symbol = alternativeSymbol;
-                return true;
-            }
-
-            alternativeSymbol = "_imp_" + symbol;
-            if (FindLibraryBySymbol(alternativeSymbol, out lib))
+            var alternativeSymbols = new [] 
```

Comment:
While this makes for nicer code, it will do a bunch of needless string allocations. I think we can easily fix by building a regex to do the matching. See https://github.com/mono/CppSharp/blob/master/src%2FGenerator%2FLibrary.cs#L129.